opendbc/safety/safety.h

```diff
@@ -211,10 +211,16 @@ bool safety_rx_hook(const CANPacket_t *to_push) {
   bool controls_allowed_prev = controls_allowed;
 
   bool valid = rx_msg_safety_check(to_push, &current_safety_config, current_hooks);
-  if (valid) {
+  bool whitelisted = get_addr_check_index(to_push, current_safety_config.rx_checks, current_safety_config.rx_checks_len) != -1;
+  if (valid && whitelisted) {
     current_hooks->rx(to_push);
   }
 
+  // the relay malfunction hook runs on all incoming rx messages
+  if (current_hooks->rx_relay_malfunction != NULL) {
+    current_hooks->rx_relay_malfunction(to_push);
+  }
```

Comment:
I don't think we want to do this. We need to keep moving towards removing the Turing completeness in the safety mode implementations. The information for this check should be fully contained the safety mode config struct and not require a full free-form C function.

Instead of adding another, I think we want to remove the fwd hook and use that same config for this. I tried this briefly but it needed some more though: https://github.com/commaai/opendbc/commit/595685a62548c50e4f9af727c3b102a2ae7f69f8.

Reply:
per usual, the right fix is likely a red diff :)